src/mlpack/core/boost_backport/boost_backport_serialization.hpp

```diff
@@ -24,5 +24,13 @@
   #include <boost/serialization/unordered_map.hpp>
 #endif
 
+#if BOOST_VERSION <= 105800
```

Comment:
Form the file:

```
/**
 * @file boost_backport.hpp
 * @author Yannis Mentekidis
 *
 * Centralized control of what boost files to include. We have backported the
 * following boost functionality here:
 *
 *  * unordered_set serialization support (added in boost 1.56.0)
 *
 * If the detected boost version is greater than 1.58.0, we include the normal
 * serialization functions (not the backported ones).  For all older versions we
 * include the backported headers.
 */
```

would be great if you could add the same for the vector backport.